src/main/java/io/javalin/apibuilder/ApiBuilder.java

```diff
@@ -486,11 +490,51 @@ public static void crud(@NotNull String path, @NotNull CrudHandler crudHandler,
         String SEPARATOR = "/:";
         String resourceBase = path.substring(0, path.lastIndexOf(SEPARATOR));
         String resourceId = path.substring(path.lastIndexOf(SEPARATOR) + SEPARATOR.length());
-        staticInstance().get(prefixPath(path), ctx -> crudHandler.getOne(ctx, ctx.pathParam(resourceId)), permittedRoles);
-        staticInstance().get(prefixPath(resourceBase), crudHandler::getAll, permittedRoles);
-        staticInstance().post(prefixPath(resourceBase), crudHandler::create, permittedRoles);
-        staticInstance().patch(prefixPath(path), ctx -> crudHandler.update(ctx, ctx.pathParam(resourceId)), permittedRoles);
-        staticInstance().delete(prefixPath(path), ctx -> crudHandler.delete(ctx, ctx.pathParam(resourceId)), permittedRoles);
+        Handler getOneHandler = ctx -> crudHandler.getOne(ctx, ctx.pathParam(resourceId));
+        Handler updateHandler = ctx -> crudHandler.update(ctx, ctx.pathParam(resourceId));
+        Handler deleteHandler = ctx -> crudHandler.update(ctx, ctx.pathParam(resourceId));
+
+        if (crudHandler instanceof DocumentedCrudHandler) {
```

Comment:
Maybe this should be moved out of `ApiBuilder`?

Reply:
I thought about this, but I couldn't think about a good solution. The only thing I can think come up with is to extract the complete crud logic into a separate file.

Can you give me an example of what you have in mind?

Reply:
> The only thing I can think come up with is to extract the complete crud logic into a separate file.

That's pretty much what I had in mind. Javalin itself shouldn't depend on the plugin, so decouple this and move it into the plugin package.

Reply:
I reduced the documentation specific code to just one line. It is unfortunately not possible to move it completely.